server/core/src/https/oauth2.rs

```diff
@@ -107,20 +107,20 @@ pub(crate) async fn oauth2_image_get(
         .await;
 
     match res {
-        Ok(image) => (
+        Ok(Some(image)) => (
             StatusCode::OK,
             [(CONTENT_TYPE, image.filetype.as_content_type_str())],
             image.contents,
         )
             .into_response(),
+        Ok(None) => {
+            warn!(?rs_name, "No image set for oauth2 client");
+            (StatusCode::NOT_FOUND, "").into_response()
```

Comment:
This could be a `Redirect` to the default image

Reply:
Can't redirect in an img tag.

Reply:
Yeah you can, works fine 😄

Reply:
Throw a HTTP 303/307 and it'll just load the file it's told to.

Reply:
In this case, I think I actually want to leae the 404. The flow is that *if* the server sends back custom-image == true, then we go to get the image, if the image doesnt exist we should 404 - it's correctly showing the image doesn't exist! 

The fallback of the default image should be processed *before* we even get to this point.